cassandra-migration/src/main/java/org/cognitor/cassandra/migration/Database.java

```diff
@@ -241,4 +297,73 @@ public Database setConsistencyLevel(ConsistencyLevel consistencyLevel) {
         this.consistencyLevel = notNull(consistencyLevel, "consistencyLevel");
         return this;
     }
+
+    /**
+     * Attempts to acquire the lead on a migration through a LightWeight
+     * Transaction.
+     *
+     * @param repositoryLatestVersion
+     *            the latest version number in the migration repository
+     * @return if taking the lead succeeded
+     */
+    boolean takeLeadOnMigrations(int repositoryLatestVersion) {
+        if (VersionNumber.parse("2.0").compareTo(cassandraVersion) > 0) {
+            // No LWT before Cassandra 2.0 so leader election can't happen
+            return true;
+        }
+
+        while (repositoryLatestVersion > getVersion()) {
+            try {
+                LOGGER.debug("Trying to take lead on schema migrations");
+                BoundStatement boundStatement = takeMigrationLeadStatement.bind(getKeyspaceName(), this.instanceId);
+                ResultSet lwtResult = session.execute(boundStatement);
+
+                if (lwtResult.wasApplied()) {
+                    LOGGER.debug("Took lead on schema migrations");
+                    tookLead = true;
+                    return true;
+                }
+
+                LOGGER.info("Schema migration is locked by another instance. Waiting for it to be released...");
+                waitFor(TAKE_LEAD_WAIT_TIME);
+            } catch (com.datastax.driver.core.exceptions.InvalidQueryException e1) {
+                // A little redundant but necessary
+                LOGGER.info("All required tables do not exist yet, waiting for them to be created...");
+                waitFor(TAKE_LEAD_WAIT_TIME);
+            }
+        }
+
+        return false;
+    }
+
+    private void waitFor(int waitTime) {
+        try {
+            Thread.sleep(waitTime);
+        } catch (InterruptedException e) {
+            throw new IllegalStateException(e);
+        }
```

Comment:
Please do let other classes in the call stack know that we were interrupted. (see [here](https://www.ibm.com/developerworks/library/j-jtp05236/index.html) for more information).

Please set the interrupted state on the current Thread before throwing the exception by calling:
```java
Thread.currentThread().interrupt();
```

Reply:
ok